File: 04-deberes/02-proyecto/Backend/zapatos/api/models/Compra.js
/**
 * Compra.js
 *
 * @description :: A model definition represents a database table/collection.
 * @docs        :: https://sailsjs.com/docs/concepts/models-and-orm/models
 */

module.exports = {

  attributes: {
    fecha: {
      type: 'string',
      required:true
    },
    cantidad: {
      type: 'number',
      min: 0,
      required:true
    },
    cantidad: {
      type: 'number',
      min: 0,
      required:true
    },
    validez:{
      type: 'boolean',
      defaultsTo: true
    },
    codigoZap: {         // Nombre del fk para la relación
      model: 'zapato',   // Nombre del modelo a relacionar (padre) 
      required: true   // OPCIONAL-> Simpre se ingrese el fk
    },
    codigoCli: {         // Nombre del fk para la relación
      model: 'cliente',   // Nombre del modelo a relacionar (padre) 
      required: true   // OPCIONAL-> Simpre se ingrese el fk
    },
   },

};

File: 04-deberes/02-proyecto/AppZapatos/app/src/main/java/com/example/appzapatos/modulo_compras/Compra.kt
package com.example.appzapatos.modulo_compras

import com.example.appzapatos.clases_auxiliares.ClienteAux
import com.example.appzapatos.clases_auxiliares.ZapatoAux

class Compra(
    var id: Int?,
    var fecha: String?,
    var cantidad: Int?,
    var total: Double?,
    var validez: Boolean?,
    var codigoCli: ClienteAux?,
    var codigoZap: ZapatoAux?

) {

}